generic_expression_patterns_modules/process.py

```diff
@@ -416,11 +416,41 @@ def compare_and_reorder_samples(expression_file, metadata_file):
 
     if metadata_sample_ids.equals(expression_sample_ids):
         print("sample ids are ordered correctly")
-        return
     else:
         # Convert gene expression ordering to be the same as
         # metadata sample ordering
         print("sample ids don't match, going to re-order gene expression samples")
         expression_data = expression_data.reindex(metadata_sample_ids)
         expression_data.to_csv(expression_file, sep="\t")
-        return
+
+
+def create_all_recount2_compendium(download_dir, output_filename):
```

Comment:
This is the function that reads each data_counts file and combine them together into compendium file.

Reply:
I need to figure out why the checks failed.

Reply:
Just curious, was there a reason to split up the downloading the concatentation step? You can probably see that when I coded it, I downloaded a file, concatenated it to a dataframe and then deleted the downloaded file. But I'm not sure how efficient that was and I think you have more experience with large data, so I was just curious to learn

Reply:
Good question. The main reason why I separated the downloading and concatenation is because R is usually slower when dealing with a lot of disk I/O, which is required in concatenation process. 

Another reason is because the downloading may take a while, so I prefer to let the downloading finish first. Originally I started the downloading on my local laptop that used WiFi. After a few hours I gave up because WiFi in my house is both slow and unstable. So I logged into my lab linux box and downloaded the recount2 data via ethernet. It was much faster (took less than 30 minutes).

Reply:
Ah..this makes sense. Thank you for the explanation